src/flags.ts

```diff
@@ -121,3 +151,70 @@ export const help = (opts: Partial<BooleanFlag<boolean>> = {}): BooleanFlag<void
     },
   })
 }
+
+type ElementType<T extends ArrayLike<unknown>> = T[number];
```

Comment:
what's the logic for `ArrayLike` vs `Array` ?  It's an es5 polyfill

Reply:
@mshanemc It's an artifact of a StackOverflow copy/paste :)

I'll update it to `ReadonlyArray`, which is really what we want here